pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -59,13 +55,13 @@ type RemoteCluster struct {
 
 // ClusterStore is a collection of clusters
 type ClusterStore struct {
-	rc        map[Metadata]*RemoteCluster
+	rc        map[string]*RemoteCluster
```

Comment:
+1 on removing Metadata - it is badly named in the first place.

Reply:
@costinm so you mean that we should make sure the secrets should always be in same namespace as istio? and +1 on this, this can make the secrets management logic more simple.

But how about handle this in another PR? We want to fix the pilot crash in this PR and can propose another PR to handle the secret namespace issue, comments?

Reply:
@costinm I opened an issue at https://github.com/istio/istio/issues/6692 to restrict cluster secrets in istio-system namespace, hope it is OK.